lib/ASTGen/Sources/ASTGen/Misc.swift

```diff
@@ -76,3 +61,120 @@ extension String {
     }
   }
 }
+
+extension ASTGenVisitor {
+  /// Obtains the start source location of the given node in the current file, skipping leading trivia.
+  @inline(__always)
+  func bridgedSourceLoc(for node: (some SyntaxProtocol)?) -> BridgedSourceLoc {
+    guard let node else {
+      return nil
+    }
+
+    return BridgedSourceLoc(at: node.positionAfterSkippingLeadingTrivia, in: self.base)
+  }
+
+  /// Given a token, obtains a uniqued `ASTContext`-owned copy of its text.
+  @inline(__always)
+  func bridgedIdentifier(for token: TokenSyntax?) -> BridgedIdentifier {
+    guard let token else {
+      return nil
+    }
+
+    var text = token.text
+    return text.withBridgedString { bridged in
+      ASTContext_getIdentifier(self.ctx, bridged)
+    }
+  }
+
+  /// Given a token, obtains a uniqued `ASTContext`-owned copy of its text, and its start source location in the
+  /// current file, skipping leading trivia.
+  @inline(__always)
+  func bridgedIdentifierAndSourceLoc(for token: TokenSyntax?) -> (BridgedIdentifier, BridgedSourceLoc) {
+    return (self.bridgedIdentifier(for: token), self.bridgedSourceLoc(for: token))
+  }
```

Comment:
I gave this quite a bit of thought and I didn’t wind up with a strong opinion, but perhaps you’re right about the calls being more idiomatic on `TokenSyntax`; it was tempting to make it a method on `ASTGenVisitor` because it only ever takes `self`.